app/src/main/java/com/thermatk/android/xf/fakegapps/FakeSignatures.java

```diff
@@ -19,30 +15,28 @@ public void handleLoadPackage(LoadPackageParam loadedPackage) {
         if (!loadedPackage.packageName.equals("android"))
             return;
 
-        final Class<?> PackageManagerService = XposedHelpers.findClass("com.android.server.pm.PackageManagerService", loadedPackage.classLoader);
+        Class<?> PackageManagerService;
+        try {
+            PackageManagerService = XposedHelpers.findClass("com.android.server.pm.PackageManagerService.ComputerEngine", loadedPackage.classLoader);
+        } catch (Exception e) {
+            PackageManagerService = XposedHelpers.findClass("com.android.server.pm.PackageManagerService", loadedPackage.classLoader);
+        }
         XposedBridge.hookAllMethods(PackageManagerService, "generatePackageInfo", new XC_MethodHook() {
             @Override
             protected void afterHookedMethod(MethodHookParam param) {
                 PackageInfo pi = (PackageInfo) param.getResult();
                 if (pi != null) {
                     String packageName = pi.packageName;
                     if (packageName.equals("com.google.android.gms") || packageName.equals("com.android.vending")) {
-                        String callerPackageName = "unknown package name";
-
-                        PackageManager pm;
-                        try {
-                            pm = AndroidAppHelper.currentApplication().getApplicationContext().getPackageManager();
-                            callerPackageName = pm.getNameForUid(Binder.getCallingUid());
-                        } catch (Exception ignored) {
-                        }
-
-                        Log.i("FakeGApps", "Returning fake signature to \"" + callerPackageName + "\"");
+                        XposedBridge.log("FakeGApps: Returning fake signature from \"" + packageName + "\"");
```

Comment:
Beware that you're now logging the package whose signature is being spoofed, which will always be either Play Services or Play Store. I think that the former message - which prints which app is retrieving the spoofed signature - could have been more useful, for example to check if signature spoofing is working for a certain app.
Also, please don't use XposedBridge.log - it will spam the Xposed log which is [highly recommended against by the Xposed docs.](https://api.xposed.info/reference/de/robv/android/xposed/XposedBridge.html)

Reply:
That code before did not work because it always logged "unknown package name" which is equalivent to "android". I think I'll just comment this out.

Reply:
Well, there's probably something wrong with the code retrieving the caller package name then.
If you have some time, could you please log the `ignored` exception in the innermost try-catch in the original code and send here the message + stack trace?
I'm curious to see what's going on, but I haven't any A12 devices here to try it out :/

Reply:
Can you try out if this APK works on Android 11? I haven't added the log message but someone reported it did not work on Android 11 after my changes:
[app-release (3).zip](https://github.com/whew-inc/FakeGApps/files/8499573/app-release.3.zip)